### PR TITLE
Write the OSS documentation ;; WLFE-29

### PR DESCRIPTION
**Pull request by ofrogon**
* Write the OSS documentation about the CLI part.
* Add templates for Issues.
* Update the dependencies in the lockfile.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/03_question.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+name: "❔ Request"
+about: A question about the tooling, a template, ...
```
Comment (patricklafrance) [inline .github/ISSUE_TEMPLATE/03_question.md:3]: Question should probably be redirected to [Github discussions](https://docs.github.com/en/discussions)

File: .github/ISSUE_TEMPLATE/01_request.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+name: "❗ Request"
+about: A request for a template change, new feature, ...
```
Comment (patricklafrance) [inline .github/ISSUE_TEMPLATE/01_request.md:3]: Request should probably be redirected to [Github discussions](https://docs.github.com/en/discussions)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+- [Babel GitHub](https://github.com/babel/babel/blob/master/doc/design/monorepo.md)
+- [Shopify GitHub](https://github.com/Shopify/quilt/blob/master/Decision%20records/00%20-%20Use%20a%20Lerna%20monorepo.md)
+- [Google](https://www.google.com/search?q=monorepo)
+
+### PNPM workspace
+
+This monorepo is using PNPM workspace feature to handle the installation of the npm dependencies and manage the packages interdependencies.
+
+It's important to note that PNPM workspace will **hoist** the npm dependencies at the root of the workspace. This means that there might not be a *node_modules* directory nested in the packages directories. The npm dependencies are installed in a *node_modules* directory at the root of the workspace and a single *pnpm-lock.yaml* file is generated at the root of the workspace.
```
Comment (patricklafrance) [inline CONTRIBUTING.md:27]: In fact, PNPM doesn't "hoist" dependencies, this is one of the nice thing about PNPM :)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+```bash
+pnpm i
+```
+
+## Develop the CLI packages
+
+The following documentation is a brief overview of the tools and processes involved in the development of the CLI packages.
+
+> To develop for each package, you can run `pnpm dev` from the project root folder. This will build all the content of the `packages` folder.
```
Comment (patricklafrance) [inline CONTRIBUTING.md:43]: I would rephrase for something like: "To start developing, open a terminal at the root of the project and execute the `pnpm dev` command. This will build all the packages and start them in "watch" mode."

Comment (patricklafrance) [reply CONTRIBUTING.md:43]: By the way, isn't the `build:dev` script missing a `--watch` option?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+When the package is built from the `pnpm dev` command, you can attach a debugger and debug directly from the TypeScript files.
+
+### Linting
+
+To run lint on the packages, call `pnpm lint` from the project root folder.
+
+### Testing
+
+Once each package is built in dev mode (`pnpm dev`), you can run the unit test using `pnpm test`.
```
Comment (patricklafrance) [inline CONTRIBUTING.md:67]: It's kind of weird to me to run `pnpm dev` before executing the tests because usually a `dev` script is started in watch mode and therefore is a long running process.

Usually, test runners are capable of building the project on their own and then execute the tests.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+To build it for development, from the folder, run `pnpm dev`. This will build the project in development mode and link the binary locally. Once done, you will be able to run it from a terminal by using the `create-project` command.
+
+When the package is built from the `pnpm dev` command, you can attach a debugger and debug directly from the TypeScript files.
+
+### Working in `@workleap/foundry`
+
+This package can be found under the folder `packages/foundry`.
+
+To build it for development, from the folder, run `pnpm dev`. This will build the project in development mode and link the binary locally. Once done, you will be able to run it from a terminal by using the `foundry` command.
```
Comment (patricklafrance) [inline CONTRIBUTING.md:57]: That would help if you could add a `foundry` command example. It could be a help command.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+
+The following documentation is a brief overview of the tools and processes involved in the development of the CLI packages.
+
+> To develop for each package, you can run `pnpm dev` from the project root folder. This will build all the content of the `packages` folder.
+
+### Working in `@workleap/create-project`
+
+This package can be found under the folder `packages/create-project`.
+
+To build it for development, from the folder, run `pnpm dev`. This will build the project in development mode and link the binary locally. Once done, you will be able to run it from a terminal by using the `create-project` command.
```
Comment (patricklafrance) [inline CONTRIBUTING.md:49]: That would help if you could add an example of a `create-project` command.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+Once each package is built in dev mode (`pnpm dev`), you can run the unit test using `pnpm test`.
+
+To run manual tests, you also have access to the `create-project` and `foundry` commands once the step above is done.
+
+### Done developing
+
+#### Method 1:
+
+You can unlink the CLI packages from their folder (`packages/create-project` and `packages/foundry`) and using `pnpm unlink`
```
Comment (patricklafrance) [inline CONTRIBUTING.md:75]: and using -> by using

File: CONTRIBUTING.md (hunk 4 above)
Comment (patricklafrance) [reply CONTRIBUTING.md:43]: After reading the "Working in `@workleap/create-project`" and "Working in `@workleap/foundry`" sections, it's not clear to me if you recommend starting both projects in dev mode from the root and if it's better to start both of them individually.

Do you have an opinion about this?

**Comment by patricklafrance** [comment]
I like the documentation good job.

The thing that bothers me thought is that if I am new to the project I won't be able to figure out how it is design and what are the important component by reading the contributing guide.

It would be interesting to add a few additional info about how the CLI engine works.

Something like:
- It's using `degit` internally to blabla..
- Templates are located in the `/templates` folder blabla..
- Every template contains tokens that are replaced based on user inputs blabla..
- etc...

File: CONTRIBUTING.md (hunk 3 above)
Comment (ofrogon) [reply CONTRIBUTING.md:27]: Yea, `link` is prbably a better choice.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ofrogon) [reply CONTRIBUTING.md:43]: I think that starting `dev` from the root can be useful in situations where you need to have both running at the same time, like "adding a template" or "adding an option".

But in other situations, it's not useful.

My personal opinion on that is if I need to run both, I will simply start them on two terminal sessions, and that's all.

File: CONTRIBUTING.md (hunk 5 above)
Comment (ofrogon) [reply CONTRIBUTING.md:67]: True, I mixed up the concepts of automatic tests (`pnpm test`) and manual testing (where `pnpm dev` is used) in this sentence... I will rewrite that part.

File: CONTRIBUTING.md (hunk 4 above)
Comment (patricklafrance) [reply CONTRIBUTING.md:43]: Got it! Make sense. I think it would be valuable to add what you described in the docs so the contributors are aware of this distinction!

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -232,4 +251,4 @@ npm *dependencies* and *peerDependencies* must be added to the package own *pack
 
 Why?
 
-Because packages hoisting is dangerous! When multiple packages of the monorepo requires the same dependencies **but with different version** there is no guarantee on which version will be hoisted to the *node_modules* directory at the root of the workspace and which version will be installed locally. To prevent all kinds of problems, always install the *devDependencies* at the root of the workspace. This ensures that every package use the same version of the dependencies.
+Because packages hoisting is dangerous! When multiple packages of the monorepo requires the same dependencies **but with different version** there is no guarantee on which version will be hoisted to the *node_modules* directory at the root of the workspace and which version will be installed locally. To prevent all kinds of problems, always install the *devDependencies* at the root of the workspace. This ensures that every package uses the same version of the dependencies.
```
Comment (patricklafrance) [inline CONTRIBUTING.md:254]: It shouldn't be an issue with PNPM since it doesn't rely on packages hoisting.

**Comment by patricklafrance** [comment]
Added a few comments but it's looking pretty good to me! GJ

File: CONTRIBUTING.md (hunk 9 above)
Comment (ofrogon) [reply CONTRIBUTING.md]: I'm not sure if it's useful to keep this all section, but I resume the idea with: `Because it's easier to maintain, update, and keep track of common tooling dependencies, all in the same file.`

**Review by patricklafrance — APPROVED** [review]
